applications/lfricinputs/fab_lfricinputs.py

```diff
@@ -110,23 +112,71 @@ def preprocess_c(self):
+                               flags=['-I$relative/include',
+                                      '-I$source/science/shumlib/common/src',
+                                      '-I$source/science/shumlib/\
+                                        shum_thread_utils/src',]),         
+                      AddFlags(match="$source/boundary_layer/*",
+                               flags=['-I$relative/include',
+                                      '-I$source/science/shumlib/common/src',
+                                      '-I$source/science/shumlib/\
+                                        shum_thread_utils/src',]),         
+                      AddFlags(match="$source/large_scale_precipitation/*",
+                               flags=['-I$relative/include',
+                                      '-I$source/science/shumlib/common/src',
+                                      '-I$source/science/shumlib/\
+                                        shum_thread_utils/src',]),         
+                      AddFlags(match="$source/free_tracers/*",
+                               flags=['-I$relative/include',
+                                      '-I$source/science/shumlib/common/src',
+                                      '-I$source/science/shumlib/\
+                                        shum_thread_utils/src',]),       
+                      # for backward compatibility
+                      AddFlags(match="$source/science/um/*",
+                               flags=['-I$relative/include',
+                                      '-I/$source/science/\
+                                        um/include/other/',
+                                      '-I$source/science/\
+                                        shumlib/common/src',
+                                      '-I$source/science/shumlib/\
+                                        shum_thread_utils/src',]),  
+                     ]
         super().preprocess_c(path_flags=path_flags)
 
     def preprocess_fortran(self):
-        path_flags = [AddFlags(match="$source/science/um/*",
-                               flags=['-I$relative/include',
-                                      '-I$source/shumlib/\
-                                       shum_thread_utils/src/']),
-                      AddFlags(match="$source/science/jules/*",
+        path_flags = [AddFlags(match="$source/science/jules/*",
                                flags=['-DUM_JULES', '-I$output']),
                       AddFlags(match="$source/science/shumlib/*",
                                flags=['-DSHUMLIB_LIBNAME=libshum',
                                       '-I$output',
                                       '-I$source/shumlib/common/src',
                                       '-I$relative'],),
                       AddFlags(match="$source/science/*",
-                               flags=['-DLFRIC'])]
+                               flags=['-DLFRIC']),
```

Comment:
Another lesson I think: maybe we can maintain the list of 'special' directories somewhere, to avoid this code duplication. Or maybe we add a special fab script to build some parts of the code that we just call? Again, nothing for this PR.